public/api-v1.php

```diff
@@ -412,6 +413,53 @@
     return $response;
 });
 
+/*********************************
+* POST /firmware
+**********************************/
+$app->post('/firmware', function(Request $request, Response $response) use ($app) {
+    $uploadedFile = array_pop($request->getUploadedFiles());
+    if ($uploadedFile->getError() === UPLOAD_ERR_OK) {
+        $uploadedFile->moveTo($this->config['rw_dir'] . 'firmware' . DIRECTORY_SEPARATOR . $uploadedFile->getClientFilename() );
```

Comment:
Before renaming the upload, we have to check if the file already exists and implement a strategy:
- returning an error?
- generating a new file name?

Elsewhere I implemented this check for the path validity

https://github.com/nethesis/tancredi/blob/1f99dd27aafa3bb58f019fc848655cc0134c1d70/public/provisioning.php#L40-L44